### PR TITLE
extend interface for grid variants

### LINKED ISSUE #139: feat: extend interface for grid variants

### PR DESCRIPTION
**Pull request by SebastianDD**
fixes #139

### CODE REVIEW

File: examples/powerfactory_control.ipynb [hunk 1]
```diff
@@ -38,12 +38,12 @@
...
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "PROJECT_NAME = \"PowerFactory-Tools\"  # may be also full path \"dir_name\\project_name\"\n",
-    "GRID_NAME = \"HV_8_Bus\"\n",
+    "PROJECT_NAME = \"PowerFactory-Tools-Dev\"  # may be also full path \"dir_name\\project_name\"\n",
```
Comment (sasanjac) [inline examples/powerfactory_control.ipynb:45]: Is this intended?

**Review by sasanjac — COMMENTED** [review]
Please change all brackets in docstrings to curly brackets

File: examples/powerfactory_control.ipynb (hunk 1 above)
Comment (SebastianDD) [reply examples/powerfactory_control.ipynb:45]: no, thats a mistake

File: powerfactory_tools/interface.py [hunk 2]
```diff
@@ -166,7 +176,7 @@ def load_powerfactory_module_from_path(self) -> PFTypes.PowerFactoryModule:
         spec.loader.exec_module(pfm)
         return t.cast("PFTypes.PowerFactoryModule", pfm)
 
-    def load_settings_dir_from_pf(self) -> PFTypes.DataDir:
+    def load_settings_dir_from_pf(self) -> PFTypes.DataObject:
```
Comment (sasanjac) [inline powerfactory_tools/interface.py:179]: whats the reasoning here?

File: powerfactory_tools/interface.py [hunk 3]
```diff
@@ -175,17 +185,28 @@ def load_settings_dir_from_pf(self) -> PFTypes.DataDir:
             raise RuntimeError(msg)
 
         logger.debug("Loading settings from PowerFactory... Done.")
-        return t.cast("PFTypes.DataDir", settings_dir)
+        return settings_dir
 
-    def load_unit_settings_dir_from_pf(self) -> PFTypes.DataDir:
+    def load_unit_settings_dir_from_pf(self) -> PFTypes.DataObject:
```
Comment (sasanjac) [inline powerfactory_tools/interface.py:190]: see above

File: powerfactory_tools/interface.py (hunk 2 above)
Comment (SebastianDD) [reply powerfactory_tools/interface.py:179]: the external data container assigned in the project settings is assumed as type DataDir

Therefore, the directories for storing the settings (units, fonts, etc.) should not be from the type DataDir as they are from different classes in PowerFactory. Either one have to use the basis class DataObject or one have to specify speparate directory types.

Comment (sasanjac) [reply powerfactory_tools/interface.py:179]: `DataDir` is a superclass for data dirs which at the moment does not differ from a `DataObject`. However, 
it is used here to indicate that its not a generic `DataObject` and may gain some additional attributes in the future.

**Comment by SebastianDD** [comment]
@sasanjac : please replace variable type t.Any

**Review by sasanjac — APPROVED** [review]

**Pull request merged**